src/modules/app/containers/Switcher.js

```diff
@@ -64,67 +64,55 @@ type PropsType = {
  */
 export class Switcher extends React.Component<PropsType> {
   /**
-   * Renders a failure component if the payload contains an error or a LoadingSpinner if the data is currently being fetched
-   * @param payload The payload to check for errors or fetching process
+   * Renders a failure component if a payload contains an error or a LoadingSpinner if the data is still being fetched
+   * @param payloads The payloads to check for errors or fetching process
    * @return {*}
    */
-  static renderFailureLoadingComponents = (payload: Payload<any>): Node => {
-    if (payload.error) {
-      return <FailureSwitcher error={payload.error} />
-    } else if (payload.isFetching || !payload.data) {
-      return <LoadingSpinner />
-    } else {
-      return null
+  static renderFailureLoadingComponents = (payloads: Array<Payload<any>>): Node => {
+    for (let i = 0; i < payloads.length; i++) {
+      const payload = payloads[i]
+      if (payload.error) {
+        return <FailureSwitcher error={payload.error} />
+      } else if (payload.isFetching || !payload.data) {
+        return <LoadingSpinner />
+      }
     }
+    return null
```

Comment:
That implementation is kind of weird I think... I suggest the following:
```jsx
const payloadWithError = payloads.find(payload => payload.error)
if (payloadWithError) {
  return <FailureSwitcher error={payloadWithError.error} />
} else if (payloads.find(payload => payload.isFetching || !payload.data)) {
  return <LoadingSpinner />
}
return null; // I'm surprised, that the type Node allows nulls...
```
(That way, we also treat displaying failures with a higher priority than displaying loadingSpinners)

Reply:
Way better than mine, thank you!

Reply:
But I even think we should display loadingSpinners with a higher priority, otherwise we sometimes display an failure for  a short time and than display something else which looks kind of weird. For example if you're on `/augsburg/xy` the categories endpoint fails instantly, whereas the cities and languages are still being loaded. This way,  an failure is displayed until these endpoints are loaded and afterwards a LanguageSwitcher is displayed.

Reply:
Hmm.. I'm not really sure, how we currently fetch stuff^^
I'd expected, that once we change our routes, all corresponding payloads are reset (if necessary), until the request is finished. Thus there shouldn't exist any endpoints that represent failures and at the same time, after waiting for other payloads are loaded, start loading themselves :thinking: (But I'm not sure tbh^^ - At least that how I thought it *should* be handled)

Reply:
No, I think you misunderstood me. We start fetching cities, languages and categories at the same time, but categories fails instantly cause `xy` is an invalid language, whereas languages and cities can be fetched.

Reply:
Hmm... I see... there are some weird dependencies there... Yeah, then we should probably do it the other way around...